RecoMET/METPUSubtraction/python/deepMETProducer_cff.py

```diff
@@ -0,0 +1,53 @@
+import FWCore.ParameterSet.Config as cms
+from FWCore.ParameterSet.pfnInPath import pfnInPath
+import os
+
+from RecoMET.METPUSubtraction.deepMETProducer_cfi import deepMETProducer as _deepMETProducer
+
+deepMETsResolutionTune = _deepMETProducer.clone()
+deepMETsResponseTune = _deepMETProducer.clone(
+    graph_path = 'RecoMET/METPUSubtraction/data/models/deepmet/deepmet_resp_v1_2018/model.graphdef',
+)
+
+from Configuration.Eras.Modifier_phase2_common_cff import phase2_common
+phase2_common.toModify(
+    deepMETsResolutionTune,
+    max_n_pf=12500,
+    graph_path="RecoMET/METPUSubtraction/data/models/deepmet_phase2/deepmet_v1_phase2/model.graphdef"
+)
+phase2_common.toModify(
+    deepMETsResponseTune,
+    max_n_pf=12500,
+    graph_path="RecoMET/METPUSubtraction/data/models/deepmet_phase2/deepmet_resp_v1_phase2/model.graphdef"
+)
+
+from Configuration.Eras.Modifier_run2_jme_2016_cff import run2_jme_2016
+run2_jme_2016.toModify(
+    deepMETsResponseTune,
+    graph_path="RecoMET/METPUSubtraction/data/models/deepmet/deepmet_resp_v1_2016/model.graphdef"
+)
+
+from RecoMET.METPUSubtraction.deepMETSonicProducer_cff import deepMETSonicProducer as _deepMETSonicProducer
+from Configuration.ProcessModifiers.deepMETSonicTriton_cff import deepMETSonicTriton
+
+# propagate possible parameter updates
+_deepMETsResolutionTuneSonic = _deepMETSonicProducer.clone(
+    max_n_pf = deepMETsResolutionTune.max_n_pf,
+    Client = dict(
+        modelName = deepMETsResolutionTune.graph_path.value().split('/')[-3],
+        modelConfigPath = '/'.join(deepMETsResolutionTune.graph_path.value().split('/')[:-2])+'/config.pbtxt',
+        modelVersion = os.path.realpath(pfnInPath(deepMETsResolutionTune.graph_path.value()).split(':')[-1]).split('/')[-2], #model "1"
```

Comment:
how about defining a function like

```python
def split_model_path(path):
    modelName = path.split('/')[-3]
    modelConfigPath = path.split('/')[:-2]
    modelVersion = os.path.realpath(pfnInPath(path).split(':')[-1]).split('/')[-2]
    return modelName, modelConfigPath, modelVersion
```
and using it in the config dicts to avoid repeated somewhat opaque string manipulation?

Reply:
Ah yes that makes sense.  I pushed a version that has a function that returns the client dictionary